nodeadm/internal/api/featuregates.go

```diff
@@ -14,6 +14,12 @@ var featureVerifiers = map[Feature]func(Feature, map[Feature]bool) bool{
 	// InstanceIdNodeNameGate controls whether to use instance ID as the node's name.
 	// By default, this feature is disabled, and the private DNS Name will be used.
 	InstanceIdNodeName: DefaultFalse,
+
+	// AggressiveImagePullTradeoff enables a parallel image pull for container
+	// images. This will use more instance CPU and Memory during image pull, but
+	// may result in faster image pull times. This flag will be ignored on
+	// instances with memory and vCPU below a certain threshold.
+	AggressiveImagePullTradeoff: DefaultFalse,
```

Comment:
(not blocking) I think we need to extend `Feature` to support additional, arbitrary requirements (like the system resources in this case). The `containerd` package doesn't seem like the best place for it IMO, but it'll work for now.

Reply:
That's fair, the way this is implemented now also repeatedly checks CPU/Memory every time we check the flag value. Having a notion of config separate from the raw config file would probably work.